Code/Framework/AzCore/Tests/Asset/BaseAssetManagerTest.h

```diff
@@ -82,4 +87,57 @@ namespace UnitTest
         AZStd::vector<AZStd::string> m_assetsWritten;
     };
 
+    
+    struct ReadRequest
+    {
+        AZStd::chrono::milliseconds m_deadline{};
+        AZ::IO::IStreamerTypes::Priority m_priority{};
+        IO::IStreamerTypes::RequestMemoryAllocatorResult m_data{ nullptr, 0, IO::IStreamerTypes::MemoryType::ReadWrite };
+        AZ::IO::IStreamer::OnCompleteCallback m_callback;
+        IO::FileRequestPtr m_request;
+    };
+
+    struct MemoryStreamerWrapper
+    {
+        MemoryStreamerWrapper();
+        ~MemoryStreamerWrapper() = default;
+
+        ReadRequest* GetReadRequest(IO::FileRequestHandle request);
+
+        template<typename TObject>
+        bool WriteMemoryFile(const AZStd::string& filePath, TObject* object, AZ::SerializeContext* context)
+        {
+            auto& buffer = m_virtualFiles[filePath];
```

Comment:
is it possible that the filePath could not be there?

Reply:
I think thats probably the only use-case right now.  The tests call WriteMemoryFile to create it.  I double checked and it does add the file just fine.